### DISCUSSION TITLE
the query_status method does not work in Win32RAW

### DISCUSSION THREAD

**Discussion by Razielthc** [python-escpos/python-escpos#656] [Q&A]
I have tested the method using Network and if it works, I would like to do the same with a printer that uses the Win32RAW connector, I checked the classes and I think the error is because Win32RAW does not have a _read() method in its class, Any suggestions on how to use that method?

from escpos.printer import Network, Usb, Win32Raw
from escpos.exceptions import DeviceNotFoundError, ConfigNotFoundError, USBNotFoundError, ConfigSectionMissingError, Error

try:
    p = Win32Raw(printer_name="POS-80C", profile="TM-T20II")

    lista = [b'\x10\x04\x01', b'\x10\x04\x02', b'\x10\x04\x03', b'\x10\x04\x04']
    for status in lista:
        respuesta = p.query_status(status)
        print("Respuesta", respuesta)

except (Exception, DeviceNotFoundError, WindowsError, ConfigNotFoundError, ConfigSectionMissingError, Error) as e:
    print("Error", type(e))

**Comment by belono** [comment]
Hi @Razielthc !

You are right, the Win32Raw connector lacks the `_read()` method.
This is because the only connectors that can really read bytes from the printer are the low level connectors (Network, Serial, Usb ...) that communicate with the printer directly over the bus. Instead, the high level connectors (Win32Raw, CupsPrinter ...) do not communicate with the printer directly but with the OS print queue, so it is necessary that the print queue exposes some kind of info that we can use to guess the printer status.

I was able to workaround this problem on the CupsPrinter connector some time ago and, while it is not perfect and the status is not accurate (it returns the print queue status, not the printer status actually), it gets the job done and maintains API compatibility with the low level connectors.

Maybe we could dig a bit into the [pywin32](https://mhammond.github.io/pywin32/win32print.html) library and try to find some method that we could use to implement the `_read()` method into the Win32Raw connector in a similar way as we did in the CupsPrinter connector.
